sources/FILEPKG

```diff
@@ -4407,87 +4407,102 @@ compiling " T)
-                      (RETURN (LISPXPRINT (CONS FILE '(not found))
-                                     T T)))
-                  [RESETSAVE NIL (LIST (FUNCTION CLOSEF?)
-                                       (SETQ FILESTREAM (OPENSTREAM FULL 'INPUT]
-                  (CL:FORMAT T "~A: " (SETQ FULL (FULLNAME FILESTREAM)))
-                  (CL:MULTIPLE-VALUE-SETQ (ENV MAP TOP)
-                         (OR (GET-ENVIRONMENT-AND-FILEMAP FULL)
-                             (\PARSE-FILE-HEADER FILESTREAM)))
-
-             (* ;; "Get reader environment of file.  The call to GET-ENVIRONMENT-AND-FILEMAP with the filename will get cached info if it exists.  Otherwise, read the top of the file")
-
-                  (SETQ RDTBL (AND ENV (fetch (READER-ENVIRONMENT REREADTABLE) of ENV)))
-                  (CL:WHEN (AND ENV (FETCH (READER-ENVIRONMENT REFORMAT) OF ENV))
-                      (\EXTERNALFORMAT FILESTREAM (FETCH (READER-ENVIRONMENT REFORMAT) OF ENV)))
-                  (SETQ CA (SEPRCASE DWIMIFYCOMPFLG RDTBL))
-                  [OR (SETQ PATTERNS (CDR (ASSOC RDTBL FFILEPOSPATTERNS)))
-                      (push FFILEPOSPATTERNS
-                            (CONS RDTBL (SETQ PATTERNS
-                                         (for ATOM in ATOMS
-                                            collect (CONCAT (COND
-                                                               ((EQ (CHCON1 ATOM)
-                                                                    (CHARCODE ESCAPE))
-                                                                (SETQ ATOM (SUBSTRING ATOM 2 -1))
-                                                                "")
-                                                               (T " "))
-                                                           [COND
-                                                              ((SETQ I (STRPOS ' ATOM))
-                                                               (SUBSTRING ATOM 1 (SUB1 I)))
-                                                              ((STRINGP ATOM))
-                                                              (T (LET ((*PACKAGE* (CL:SYMBOL-PACKAGE
-                                                                                   ATOM)))
+        (CL:UNLESS 
+            [NLSETQ
+             (RESETLST
+                 [PROG (PATTERNS CA RDTBL MAP FILESTREAM PRINTFLG ENV TOP I)
+                       (OR (SETQ FULL (FINDFILE FILE))
+                           (RETURN (LISPXPRINT (CONS FILE '(not found))
+                                          T T)))
+                       [RESETSAVE NIL (LIST (FUNCTION CLOSEF?)
+                                            (SETQ FILESTREAM (OPENSTREAM FULL 'INPUT]
+                       (CL:FORMAT T "~A: " FULL)
+                       (CL:MULTIPLE-VALUE-SETQ (ENV MAP TOP)
+                              (OR (GET-ENVIRONMENT-AND-FILEMAP FILESTREAM)
```

Comment:
The problem was that loading the filemap would cause the file to be closed.  So the point of this is to make sure that it is reopened after the filemap is loaded.